servermaster/executor_manager.go

```diff
@@ -103,7 +103,8 @@ func (e *ExecutorManagerImpl) HandleHeartbeat(req *pb.HeartbeatRequest) (*pb.Hea
 	exec.heartbeatTTL = time.Duration(req.Ttl) * time.Millisecond
 	exec.Status = model.ExecutorStatus(req.Status)
 	usage := model.RescUnit(req.GetResourceUsage())
-	err := e.rescMgr.Update(exec.ID, usage, exec.Status)
+	// TODO: update reserve resources by heartbeats.
+	err := e.rescMgr.Update(exec.ID, usage, usage, exec.Status)
```

Comment:
Does `reserve` mean the resource manager allocates some resource (the resource can be either used or unused but prepare to be used)

Reply:
Right, the resource manager allows the tasks to use this amount of resource. The `Reserved` resource is estimated to be used, same as the `Cost` of tasks. If we use a container to run a task, we can limit the resource exactly by the `Reserved` amount.

Reply:
However, things always go sideway. In the go (or java like what spark runs on) runtime, the tasks can use much more (or less) resources than expected. In this case, there might be a global resource pool that allows the tasks to alloc more resource than the reserved. If the global resource pool is exhuasted, then oom occures.

Reply:
why use the same value here